src/server/scripts/Spells/spell_monk.cpp

```diff
@@ -159,9 +164,63 @@ class spell_monk_provoke : public SpellScript
     }
 };
 
+// Roll - 109132 or Roll (3 charges) - 121827
+class spell_monk_roll : public SpellScriptLoader
+{
+  public:
+    spell_monk_roll() : SpellScriptLoader("spell_monk_roll") { }
+
+    class spell_monk_roll_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_monk_roll_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo({ SPELL_MONK_ROLL });
+        }
+
+        void HandleBeforeCast()
+        {
+            Aura* aur = GetCaster()->AddAura(SPELL_MONK_ROLL_TRIGGER, GetCaster());
+            if (!aur)
+                return;
+
+            AuraApplication* app = aur->GetApplicationOfTarget(GetCaster()->GetGUID());
+            if (!app)
+                return;
+
+            app->ClientUpdate();
+        }
+
+        void HandleAfterCast()
+        {
+            Unit* caster = GetCaster();
+            if (!caster || caster->GetTypeId() != TYPEID_PLAYER)
+                return;
+
+            caster->CastSpell(caster, SPELL_MONK_ROLL_TRIGGER, true);
+
+            if (caster->HasAura(SPELL_MONK_ITEM_PVP_GLOVES_BONUS))
+                caster->RemoveAurasByType(SPELL_AURA_MOD_DECREASE_SPEED);
```

Comment:
Shouldn't this be done by casting 125127 instead of manually removing aura type?

Reply:
Actually, no item has this effect attached to it, why is it here at all